### PR TITLE
Slice tools

### PR DESCRIPTION
**Pull request by LydiaFrance**
I have removed the utils.py and instead created a couple of modules. 

- create_mock_data
 This will become useful as a module because we also need some create mock svd outputs and mock U/t/X arrays. 

- slice_tools
This will become useful as i add functionality for loading the saved slice files. 

# Copilot's Summary:

This pull request refactors the `dmd_era5` package by reorganizing utility functions into more specific modules and adding tests for the new modules. The main changes include creating new modules for mock data generation and slicing tools, and updating the imports and tests accordingly.

### Codebase Reorganization:

* Moved `create_mock_era5` function to a new module `dmd_era5.create_mock_data.create_mock_data` and added logging. (`src/dmd_era5/create_mock_data/create_mock_data.py`)
* Moved `slice_era5_dataset`, `thin_era5_dataset`, and `standardize_data` functions to a new module `dmd_era5.slice_tools.slice_tools` and added logging. (`src/dmd_era5/slice_tools/slice_tools.py`)

### Import Updates:

* Updated imports in `src/dmd_era5/__init__.py` to reflect the new module structure. (`src/dmd_era5/__init__.py`) [[1]](diffhunk://#diff-6a74c7a9e542fcaa49f56960f371edfb135e7d76a33ef67ffbdcc4fde4b6b445L11-R12) [[2]](diffhunk://#diff-6a74c7a9e542fcaa49f56960f371edfb135e7d76a33ef67ffbdcc4fde4b6b445R21)
* Added `__init__.py` files for the new modules to expose the functions. (`src/dmd_era5/create_mock_data/__init__.py`, `src/dmd_era5/slice_tools/__init__.py`) [[1]](diffhunk://#diff-8d3746144c3e9ad4874d29a5b8efb58cf220879c64621272bc604af5fbc6aa83R1-R5) [[2]](diffhunk://#diff-bae6e6d1f6a6b89e03ca4626031b92d46ed05a7c077231d099f7c4e9de65ec75R1-R11)

### Test Updates:

* Created new test file `tests/test_01_create_mock_data.py` for the `create_mock_data` module. (`tests/test_01_create_mock_data.py`)
* Removed the old test file `tests/test_01_utils.py` and its tests, as they have been redistributed to the new modules. (`tests/test_01_utils.py`)

### Code Cleanup:

* Removed the old `utils.py` file, which previously contained the refactored functions. (`src/dmd_era5/utils.py`)

### CODE REVIEW

**Comment by LydiaFrance** [comment]
Note that I tried to add some more testing for standardization, but was unable to get it to work. These tests are commented out for now. #11

**Review by dsj976 — CHANGES_REQUESTED** [review]
Hi have modified the `standardize_data` function so that the mean is always removed, regardless of whether the user requests to scale the data to unit variance or not. I don't think we would ever be in a situation where we would want to scale the data and not remove the mean (not sure if that even makes sense statistically). I think it would also make sense to rename the `slice_tools.py` module to `data_utils`.py (or something similar), as not all the functions in `slice_tools.py` are meant for slicing or thinning datasets. There might be a better name than `data_utils.py`, happy to hear suggestions from you @LydiaFrance @louisavz.

**Comment by LydiaFrance** [comment]
Noted, @dsj976 -- I think in this case "slice" means the noun rather than the verb, it refers to the slice of ERA5 that has been downloaded and now needs to be loaded/sliced further (as in the SVD step)/thinned/standardised etc. "Data" is also a little too broad in this case, we will have svd results saved too which are different in this case. 

I know you like the term utils because it is a catch all, but that is why it is not a good name for the negative case -- it isn't clear what would NOT belong in utils.

File: tests/test_01_create_mock_data.py
Comment (louisavz) [inline tests/test_01_create_mock_data.py:21]: Recommend printing out some error message if it makes sense, although perhaps not necessary since this is under the test module... e.g.,

assert (
    isinstance(mock_ds, xr.Dataset),
    f"moc_ds is not of an XR.Dataset instance, but of type {type(mock_ds)}"
)

File: src/dmd_era5/slice_tools/slice_tools.py
Comment (louisavz) [inline src/dmd_era5/slice_tools/slice_tools.py:124]: subsampling_era5_dataset?

**Comment by louisavz** [comment]
What are some functions that don't belong to slice_tools? I agree with you both that there might be a better name than "data_utils", the "data" part is a bit broad.

File: tests/test_01_create_mock_data.py
Comment (louisavz) [reply tests/test_01_create_mock_data.py:21]: Apologies, I should have started a review instead of adding as a single comment!

Comment (dsj976) [reply tests/test_01_create_mock_data.py:21]: Yes, that's good practice and it's what we have been doing in most (although apparently not all) `assert` statements

File: src/dmd_era5/slice_tools/slice_tools.py
Comment (dsj976) [reply src/dmd_era5/slice_tools/slice_tools.py:124]: yes, I agree, because now we are using the `resample` function instead of `thin`

**Comment by dsj976** [comment]
> What are some functions that don't belong to slice_tools?

I think for now all of the functions in `slice_tools.py` are fine there. Might need to shuffle things around a bit as we add more capability

File: tests/test_01_create_mock_data.py
Comment (dsj976) [reply tests/test_01_create_mock_data.py]: Done

File: src/dmd_era5/slice_tools/slice_tools.py
Comment (dsj976) [reply src/dmd_era5/slice_tools/slice_tools.py]: done

**Review by dsj976 — APPROVED** [review]

**Comment by dsj976** [comment]
@LydiaFrance @louisavz - I have set up some protection rules of the `main` branch, and now need your approval in order to merge because I was the last pusher. When you have some time, please review, approve if you are happy, and merge 🚀

**Comment by LydiaFrance** [comment]
I think because it's still my pull request I can't explictly review and approve. However I have and I very much approve! Thank you for those changes @dsj976 and such helpful comments @louisavz

**Review by louisavz — APPROVED** [review]

**Pull request merged**
